torch/csrc/jit/mobile/model_compatibility.cpp

```diff
@@ -100,5 +102,73 @@ int64_t _get_model_bytecode_version(std::vector<IValue> bytecode_ivalues) {
   return -1;
 }
 
+/********************** Operators and Info **********************/
+
+// Forward declare
+std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
+    std::vector<IValue> bytecode_ivalues);
+
+std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
+    std::istream& in) {
+  std::unique_ptr<IStreamAdapter> rai = std::make_unique<IStreamAdapter>(&in);
+  return _get_model_ops_and_info(std::move(rai));
+}
+
+std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
+    const std::string& filename) {
+  std::unique_ptr<FileAdapter> rai = std::make_unique<FileAdapter>(filename);
+  return _get_model_ops_and_info(std::move(rai));
+}
+
+std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
+    std::shared_ptr<ReadAdapterInterface> rai) {
+  if (!check_zip_file(rai)) {
+    TORCH_WARN("Failed to open zip file for model ops.");
+    return std::unordered_map<std::string, OperatorInfo>{};
+  }
+  PyTorchStreamReader reader(std::move(rai));
+  auto bytecode_values = get_bytecode_values(reader);
+  return _get_model_ops_and_info(bytecode_values);
+}
+
+std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
+    std::vector<IValue> bytecode_ivalues) {
+  if (_get_model_bytecode_version(bytecode_ivalues) < 6) {
+    TORCH_WARN(
+        "This model likely has an older bytecode version without operator schema information. Please re-export your model to generate it");
+  }
+  std::unordered_map<std::string, OperatorInfo> result;
+  if (!bytecode_ivalues.empty()) {
+    // loop over all the functions in the bytecode
+    for (int i = 1; i < bytecode_ivalues.size(); i++) {
+      // descend to the operators list
+      auto method_tuple = bytecode_ivalues[i].toTuple()->elements();
+      auto operators_tuple = method_tuple[1].toTuple()->elements()[1];
+      auto operators = operators_tuple.toTuple()->elements()[1];
+      for (auto& op_tuple : operators.toTuple()->elements()) {
+        auto op = op_tuple.toTuple()->elements();
+
+        // grab name
+        std::string op_name = op[0].toStringRef();
+        std::string op_overload_name = op[1].toStringRef();
+        if (op_overload_name != "") {
+          op_name.append(".");
+          op_name.append(op_overload_name);
+        }
+
+        // grab schema
+        int num_args = -1;
+        if (op.size() > 2) {
+          num_args = op[2].toInt();
+        }
+        result.emplace(op_name, OperatorInfo{num_args});
```

Comment:
-1 args in the schema isnt possible for a normal operator. I'm essentially using it as a null value so that this method can still be run on models from old bytecode versions and still get a useable result back. i.e. you can still at least get the model ops, but the schema information will be empty.

Reply:
maybe just add a comment about it